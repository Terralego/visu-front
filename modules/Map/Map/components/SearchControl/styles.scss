@import "node_modules/@blueprintjs/core/lib/scss/variables";

.mapboxgl-ctrl-search {
  position: relative;
  overflow: visible;

  &__input {
    position: absolute;
    top: 0;
    right: 0;
    width: 0;
    transition: width .2s ease-in;
    overflow: hidden;
    border-radius: 4px;
    box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.4);

    &--expanded {
      width: 20rem;
    }

    .bp3-input:focus {
      box-shadow: none;
    }
  }

  .search-input {
    padding: 0;
  }
  .search-results {
    padding: 0;
    border-radius: 0 0 4px 4px;
    z-index: 1;
    position: relative;

    &__title {
      padding: .5rem .7rem;
      border-bottom: 1px solid $pt-dark-divider-black;
    }
  
    &__group {
      .bp3-divider {
        border-color: $pt-dark-divider-white;
        margin: 0;
      }
    }

    &__group-title,
    &__item {
      padding: .5rem .7rem;
      margin: 0;
    }

    &__list {
      margin: 0;
      padding: 0;
      list-style: none;
    }

    &__item {
      &--active {
        padding: 0;
      }

      button {
        cursor: pointer;
        border: 0;
        background: inherit;
        color: inherit;
        padding: .5rem .7rem;
        width: 100%;
        text-align: left;
      }

      &--active:hover,
      &--selected {
        background-color: $pt-intent-primary;
      }
    }
  }
}
