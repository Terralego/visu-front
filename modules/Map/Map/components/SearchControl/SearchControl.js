import React from 'react';
import ReactDOM from 'react-dom';
import classnames from 'classnames';
import { Icon } from '@blueprintjs/core';
import debounce from 'lodash.debounce';

import SearchInput from './SearchInput';

import translateMock from '../../../../../utils/translate';

import './styles.scss';

export class SearchControl extends React.Component {
  state = {
    visible: false,
    expanded: false,
    query: '',
    results: null,
    selected: -1,
  };

  debouncedSearch = debounce(() => this.search(), 200);

  componentDidMount () {
    global.addEventListener('click', this.listener = e => {
      const { container } = this.props;
      const { query } = this.state;

      if (query || container.contains(e.target)) return;

      this.toggle(false);
    });
  }

  componentWillUnmount () {
    global.removeEventListener('click', this.listener);
  }

  onAdd (map) {
    this.map = map;
    this.container = document.createElement('div');
    this.container.className = 'mapboxgl-ctrl mapboxgl-ctrl-group mapboxgl-ctrl-search';
    ReactDOM.render(<SearchControl container={this.container} {...this.props} />, this.container);
    return this.container;
  }

  onRemove () {
    this.container.parentNode.removeChild(this.container);
    delete this.map;
  }

  get flatResults () {
    const { results } = this.state;
    return results && results.reduce((all, { results: groupResults }) => [
      ...all,
      ...groupResults,
    ], []);
  }

  toggle = state => this.setState(({ visible }) => {
    if (visible && state !== true) {
      setTimeout(() => this.setState({ visible: false }), 500);
      return ({ expanded: false, results: null, selected: -1, query: '' });
    }
    if (!visible && state !== false) {
      setTimeout(() => {
        const { container } = this.props;
        this.setState({ expanded: true });
        container.querySelector('input').focus();
      });
      return ({ visible: true });
    }
    return null;
  });

  close = () => this.toggle(false);

  onChange = ({ target: { value: query } }) => {
    this.setState({ query });
    this.debouncedSearch();
  }

  search = async () => {
    const { onSearch } = this.props;
    const { query } = this.state;

    if (query.length < 3) {
      this.setState({ results: null, selected: -1 });
      return;
    }

    const results = await onSearch(query, this.map);

    this.setState({ results: results || true });
  }

  onKeyPress = ({ key }) => {
    const results = this.flatResults;
    if (!results) return;

    if (key === 'Escape') {
      this.toggle(false);
    }
    if (key === 'ArrowDown') {
      this.selectResultItem(+1);
    }
    if (key === 'ArrowUp') {
      this.selectResultItem(-1);
    }
    if (key === 'Enter') {
      const { selected } = this.state;
      const selectedResult = results[selected];
      if (!selectedResult) return;
      const { onResultClick } = this.props;
      onResultClick(selectedResult);
    }
  }

  selectResultItem (dir) {
    this.setState(({ selected }) => {
      const max = this.flatResults.length;

      if (selected + dir < 0) {
        return { selected: max - 1 };
      }

      if (selected + dir >= max) {
        return { selected: 0 };
      }

      return { selected: selected + dir };
    });
  }

  render () {
    const {
      renderSearchResults: SearchResults,
      onResultClick,
      translate = translateMock,
    } = this.props;
    const { visible, expanded, query, results, selected } = this.state;

    return (
      <>
        <button
          className="mapboxgl-ctrl-icon"
          type="button"
          title={translate('terralego.map.search_control.button_label')}
          aria-label={translate('terralego.map.search_control.button_label')}
          onClick={this.toggle}
        >
          <Icon icon="search" />
        </button>
        {visible && (
          <div className={classnames({
            'mapboxgl-ctrl-search__input': true,
            'mapboxgl-ctrl-search__input--expanded': expanded,
          })}
          >
            <SearchInput
              {...this.props}
              onChange={this.onChange}
              onClose={this.close}
              query={query}
              onKeyPress={this.onKeyPress}
            />
            {results && (
              <SearchResults
                {...this.props}
                results={results}
                onClick={onResultClick}
                selected={this.flatResults[selected]}
              />
            )}
          </div>
        )}
      </>
    );
  }
}

export default SearchControl;
