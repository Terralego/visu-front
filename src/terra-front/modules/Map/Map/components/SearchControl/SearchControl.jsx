import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Icon } from '@blueprintjs/core';
import debounce from 'lodash.debounce';
import SearchInput from './SearchInput';

import AbstractMapControl from '../../../helpers/AbstractMapControl';
import translateMock from '../../../../../utils/translate';
import Tooltip from '../../../../../components/Tooltip';

import './styles.scss';

export class SearchControl extends AbstractMapControl {
  static containerClassName = 'mapboxgl-ctrl mapboxgl-ctrl-group mapboxgl-ctrl-search';

  static propTypes = {
    /** Function called when user submit input. Takes query as parameter */
    onSearch: PropTypes.func,
    /** Function called when user click on a result item. Takes result object as parameter */
    onResultClick: PropTypes.func,
    /** Function used to render the search results. Default to bundled component */
    renderSearchResults: PropTypes.func,
    /** Function used to translate wording. Takes key and object of options as parameters */
    translate: PropTypes.func,
  }

  static defaultProps = {
    onSearch () {},
    onResultClick () {},
    translate: translateMock({
      'terralego.map.search_control.button_label': 'Search',
    }),
  }

  state = {
    visible: false,
    expanded: false,
    query: '',
    results: null,
    displayResults: false,
    selected: -1,
  };

  debouncedSearch = debounce(() => this.search(), 200);

  componentDidMount () {
    this.listener = ({ target }) => {
      const { container } = this.props;
      const { query } = this.state;

      if (container.contains(target)) return;

      this.toggleResultsDisplay(false);

      if (query) return;

      this.toggle(false);
    };
    global.addEventListener('click', this.listener);
  }

  componentWillUnmount () {
    global.removeEventListener('click', this.listener);
    this.isUnmount = true;
  }

  get flatResults () {
    const { results } = this.state;
    return Array.isArray(results) && results.reduce((all, { results: groupResults }) => [
      ...all,
      ...groupResults,
    ], []);
  }

  toggle = state => this.setState(({ visible }) => {
    if (visible && state !== true) {
      // SetTimeout because of css animation
      setTimeout(() => {
        if (this.isUnmount) return;
        this.setState({ visible: false });
      }, 500);
      return ({ expanded: false, results: null, selected: -1, query: '' });
    }
    if (!visible && state !== false) {
      setTimeout(() => {
        if (this.isUnmount) return;
        const { container } = this.props;
        this.setState({ expanded: true });
        container.querySelector('input').focus();
      });
      return ({ visible: true });
    }
    return null;
  });

  close = () => this.toggle(false);

  onChange = ({ target: { value: query } }) => {
    this.setState({ query });
    this.debouncedSearch();
  }

  onKeyPress = ({ key }) => {
    const results = this.flatResults;

    this.toggleResultsDisplay(true);

    if (!results) return;

    if (key === 'Escape') {
      this.toggle(false);
    }
    if (key === 'ArrowDown') {
      this.selectResultItem(+1);
    }
    if (key === 'ArrowUp') {
      this.selectResultItem(-1);
    }
    if (key === 'Enter') {
      const { selected } = this.state;
      const selectedResult = results[selected];
      if (!selectedResult) return;
      this.clickOnResult(selectedResult);
    }
  }

  toggleResultsDisplay = state => this.setState({ displayResults: state });

  clickOnResult = result => {
    const { onResultClick } = this.props;
    onResultClick({
      result,
      setQuery: query => this.setState({ query }),
    });
    this.toggleResultsDisplay(false);
  }

  async search () {
    const { onSearch } = this.props;
    const { query } = this.state;

    if (query.length < 3) {
      this.setState({ displayResults: false, results: null, selected: -1 });
      return;
    }

    this.setState({ loading: true });

    const results = await onSearch(query, this.map);

    if (this.isUnmount) return;
    this.setState({ displayResults: !!results, results, loading: false });
  }

  selectResultItem (dir) {
    this.setState(({ selected }) => {
      const max = this.flatResults.length;

      if (selected + dir < 0) {
        return { selected: max - 1 };
      }

      if (selected + dir >= max) {
        return { selected: 0 };
      }

      return { selected: selected + dir };
    });
  }

  render () {
    const {
      renderSearchResults: SearchResults,
      translate,
    } = this.props;
    const { visible, expanded, query, displayResults, results, selected, loading } = this.state;

    return (
      <>
        <Tooltip
          content={translate('terralego.map.search_control.button_label')}
        >
          <button
            className="mapboxgl-ctrl-icon"
            type="button"
            aria-label={translate('terralego.map.search_control.button_label')}
            onClick={this.toggle}
          >
            <Icon icon="search" />
          </button>
        </Tooltip>
        {visible && (
          <div className={classnames({
            'mapboxgl-ctrl-search__input': true,
            'mapboxgl-ctrl-search__input--expanded': expanded,
          })}
          >
            <SearchInput
              {...this.props}
              onChange={this.onChange}
              onClose={this.close}
              onFocus={() => this.toggleResultsDisplay(true)}
              query={query}
              onKeyPress={this.onKeyPress}
              loading={loading}
            />
            {displayResults && Array.isArray(results) && (
              <SearchResults
                {...this.props}
                results={results}
                onClick={this.clickOnResult}
                selected={this.flatResults[selected]}
              />
            )}
          </div>
        )}
      </>
    );
  }
}

export default SearchControl;
