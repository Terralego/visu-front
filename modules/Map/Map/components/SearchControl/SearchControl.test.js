import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SearchControl from './SearchControl';

jest.mock('lodash.debounce', () => fn => () => fn());
jest.useFakeTimers();

it('should render', () => {
  const tree = renderer.create(
    <SearchControl
      renderSearchResults={props => <div {...props}>SearchResults</div>}
    />,
  );
  tree.getInstance().setState({
    visible: true,
    results: [{
      group: 'foo',
      results: [{
        label: 'foo1',
      }, {
        label: 'foo2',
      }],
    }, {
      group: 'bar',
      results: [],
    }],
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should be added on map', () => {
  const instance = new SearchControl();
  const map = {};
  jest.spyOn(ReactDOM, 'render');
  instance.onAdd(map);
  expect(instance.map).toBe(map);
  expect(instance.container.constructor).toBe(HTMLDivElement);
  expect(instance.container.className).toBe('mapboxgl-ctrl mapboxgl-ctrl-group mapboxgl-ctrl-search');
  expect(ReactDOM.render).toHaveBeenCalled();
  expect(ReactDOM.render.mock.calls[0][1]).toBe(instance.container);
});

it('should be removed from map', () => {
  const instance = new SearchControl();
  jest.spyOn(ReactDOM, 'render');
  instance.container = {
    parentNode: {
      removeChild: jest.fn(),
    },
  };
  instance.onRemove();
  expect(instance.container.parentNode.removeChild).toHaveBeenCalledWith(instance.container);
  expect(instance.map).not.toBeDefined();
});

it('should add listener on mount', () => {
  jest.spyOn(global, 'addEventListener');
  const wrapper = shallow(<SearchControl />);
  const instance = wrapper.instance();
  expect(global.addEventListener).toHaveBeenCalledWith('click', instance.listener);
});

it('should close control on click', () => {
  jest.spyOn(global, 'addEventListener');
  const wrapper = shallow(<SearchControl />);
  const instance = wrapper.instance();
  instance.toggle = jest.fn();

  instance.props = {
    container: {
      contains: () => true,
    },
  };
  instance.listener({ target: null });
  expect(instance.toggle).not.toHaveBeenCalled();
  instance.toggle.mockClear();

  instance.props = {
    container: {
      contains: () => false,
    },
  };
  instance.state = {
    query: 'abc',
  };
  instance.listener({ target: null });
  expect(instance.toggle).not.toHaveBeenCalled();
  instance.toggle.mockClear();

  instance.props = {
    container: {
      contains: () => false,
    },
  };
  instance.state = {};
  instance.listener({ target: null });
  expect(instance.toggle).toHaveBeenCalled();
});

it('should remove listener on unmount', () => {
  const instance = new SearchControl();
  instance.listener = {};
  jest.spyOn(global, 'removeEventListener');
  instance.componentWillUnmount();
  expect(global.removeEventListener).toHaveBeenCalledWith('click', instance.listener);
});

it('should debounce search', () => {
  const instance = new SearchControl();
  instance.search = jest.fn();
  instance.debouncedSearch();
  expect(instance.search).toHaveBeenCalled();
});

it('should get flat results', () => {
  const results = [{
    group: 'foo',
    results: [{
      label: 'foo1',
    }, {
      label: 'foo2',
    }],
  }, {
    group: 'bar',
    results: [{
      label: 'bar1',
    }, {
      label: 'bar2',
    }],
  }];
  const instance = new SearchControl({ });
  instance.state = {
    results,
  };
  expect(instance.flatResults).toEqual([{
    label: 'foo1',
  }, {
    label: 'foo2',
  }, {
    label: 'bar1',
  }, {
    label: 'bar2',
  }]);
});

it('should toggle results', () => {
  const instance = new SearchControl();
  let stateCallback;
  instance.setState = jest.fn(fn => { stateCallback = fn; });
  instance.toggle();
  expect(instance.setState).toHaveBeenCalled();

  instance.setState = jest.fn();
  expect(stateCallback({ visible: true })).toEqual({ expanded: false, results: null, selected: -1, query: '' });
  jest.runAllTimers();
  expect(instance.setState).toHaveBeenCalledWith({ visible: false });

  instance.setState = jest.fn();
  const inputMock = {
    focus: jest.fn(),
  };
  instance.props = {
    container: {
      querySelector: jest.fn(() => inputMock),
    },
  };
  expect(stateCallback({ visible: false })).toEqual({ visible: true });
  jest.runAllTimers();
  expect(instance.setState).toHaveBeenCalledWith({ expanded: true });
  expect(instance.props.container.querySelector).toHaveBeenCalledWith('input');
  expect(inputMock.focus).toHaveBeenCalled();
});

it('should force toggle results', () => {
  const instance = new SearchControl();
  let stateCallback;
  instance.setState = jest.fn(fn => { stateCallback = fn; });

  instance.toggle(false);
  expect(stateCallback({ visible: false })).toEqual(null);
  expect(stateCallback({ visible: true })).toEqual({ expanded: false, results: null, selected: -1, query: '' });

  instance.toggle(true);
  expect(stateCallback({ visible: false })).toEqual({ visible: true });
  expect(stateCallback({ visible: true })).toEqual(null);
});

it('should close results', () => {
  const instance = new SearchControl();
  instance.toggle = jest.fn();
  instance.close();
  expect(instance.toggle).toHaveBeenCalledWith(false);
});

it('should handle input value change', () => {
  const instance = new SearchControl();
  instance.setState = jest.fn();
  instance.debouncedSearch = jest.fn();
  instance.onChange({ target: { value: 'foo' } });
  expect(instance.setState).toHaveBeenCalledWith({ query: 'foo' });
  expect(instance.debouncedSearch).toHaveBeenCalled();
});

it('should handle key press', () => {
  const onResultClick = jest.fn();
  const instance = new SearchControl({ onResultClick });
  const selected = {
    label: 'foo',
  };
  instance.state = {
    results: [{
      results: [selected],
    }],
    selected: 0,
  };
  instance.toggle = jest.fn();
  instance.selectResultItem = jest.fn();

  instance.onKeyPress({ key: 'a' });
  expect(onResultClick).not.toHaveBeenCalled();
  expect(instance.toggle).not.toHaveBeenCalled();
  expect(instance.selectResultItem).not.toHaveBeenCalled();

  instance.onKeyPress({ key: 'Escape' });
  expect(onResultClick).not.toHaveBeenCalled();
  expect(instance.toggle).toHaveBeenCalledWith(false);
  expect(instance.selectResultItem).not.toHaveBeenCalled();
  instance.toggle.mockClear();

  instance.onKeyPress({ key: 'ArrowDown' });
  expect(onResultClick).not.toHaveBeenCalled();
  expect(instance.toggle).not.toHaveBeenCalled();
  expect(instance.selectResultItem).toHaveBeenCalledWith(1);
  instance.selectResultItem.mockClear();

  instance.onKeyPress({ key: 'ArrowUp' });
  expect(onResultClick).not.toHaveBeenCalled();
  expect(instance.toggle).not.toHaveBeenCalled();
  expect(instance.selectResultItem).toHaveBeenCalledWith(-1);
  instance.selectResultItem.mockClear();

  instance.onKeyPress({ key: 'Enter' });
  expect(onResultClick).toHaveBeenCalledWith(selected);
  expect(instance.toggle).not.toHaveBeenCalled();
  expect(instance.selectResultItem).not.toHaveBeenCalled();
  onResultClick.mockClear();

  instance.state.results = [{
    results: [],
  }];

  instance.onKeyPress({ key: 'Enter' });
  expect(onResultClick).not.toHaveBeenCalled();
  expect(instance.toggle).not.toHaveBeenCalled();
  expect(instance.selectResultItem).not.toHaveBeenCalled();

  instance.state.results = null;

  instance.onKeyPress({ key: 'Enter' });
  expect(onResultClick).not.toHaveBeenCalled();
  expect(instance.toggle).not.toHaveBeenCalled();
  expect(instance.selectResultItem).not.toHaveBeenCalled();
});

it('should search', async () => {
  const results = [];
  const onSearch = jest.fn(() => results);
  const instance = new SearchControl({ onSearch });
  instance.map = {};
  instance.setState = jest.fn();
  instance.state = {};

  instance.state.query = 'ab';
  await instance.search();
  expect(instance.setState).toHaveBeenCalledWith({ results: null, selected: -1 });

  instance.state.query = 'abc';
  await instance.search();
  expect(onSearch).toHaveBeenCalledWith('abc', instance.map);
  expect(instance.setState).toHaveBeenCalledWith({ results });

  instance.state.query = 'abc';
  instance.props.onSearch = jest.fn();
  await instance.search();
  expect(onSearch).toHaveBeenCalledWith('abc', instance.map);
  expect(instance.setState).toHaveBeenCalledWith({ results: true });
});

it('should select result item', () => {
  const instance = new SearchControl();
  instance.state = {
    results: [{
      results: [{
        label: '1',
      }, {
        label: '2',
      }],
    }, {
      results: [{
        label: '3',
      }, {
        label: '4',
      }],
    }],
  };
  let stateCallback;
  instance.setState = jest.fn(fn => { stateCallback = fn; });
  instance.selectResultItem(1);
  expect(stateCallback({ selected: -1 })).toEqual({ selected: 0 });
  expect(stateCallback({ selected: 0 })).toEqual({ selected: 1 });
  expect(stateCallback({ selected: 1 })).toEqual({ selected: 2 });
  expect(stateCallback({ selected: 2 })).toEqual({ selected: 3 });
  expect(stateCallback({ selected: 3 })).toEqual({ selected: 0 });

  instance.selectResultItem(-1);
  expect(stateCallback({ selected: -1 })).toEqual({ selected: 3 });
  expect(stateCallback({ selected: 3 })).toEqual({ selected: 2 });
  expect(stateCallback({ selected: 2 })).toEqual({ selected: 1 });
  expect(stateCallback({ selected: 1 })).toEqual({ selected: 0 });
  expect(stateCallback({ selected: 0 })).toEqual({ selected: 3 });
});
