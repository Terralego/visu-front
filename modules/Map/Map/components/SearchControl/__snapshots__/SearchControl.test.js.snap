// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <button
    aria-label="terralego.map.search_control.button_label"
    className="mapboxgl-ctrl-icon"
    onClick={[Function]}
    title="terralego.map.search_control.button_label"
    type="button"
  >
    <span
      className="bp3-icon bp3-icon-search"
      icon="search"
    >
      <svg
        data-icon="search"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          search
        </desc>
        <path
          d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </button>,
  <div
    className="mapboxgl-ctrl-search__input"
  >
    <div
      className="bp3-card bp3-elevation-0 search-input"
    >
      <div
        className="bp3-input-group"
      >
        <input
          className="bp3-input"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "paddingRight": 10,
            }
          }
          type="text"
          value=""
        />
        <span
          className="bp3-input-action"
        >
          <button
            className="bp3-button"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              className="bp3-icon bp3-icon-small-cross"
              icon="small-cross"
            >
              <svg
                data-icon="small-cross"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <desc>
                  small-cross
                </desc>
                <path
                  d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </div>
    </div>
    <div
      onClick={[Function]}
      renderSearchResults={[Function]}
      results={
        Array [
          Object {
            "group": "foo",
            "results": Array [
              Object {
                "label": "foo1",
              },
              Object {
                "label": "foo2",
              },
            ],
          },
          Object {
            "group": "bar",
            "results": Array [],
          },
        ]
      }
    >
      SearchResults
    </div>
  </div>,
]
`;
